File: 965.univalued-binary-tree.java
/*
 * @lc app=leetcode id=965 lang=java
 *
 * [965] Univalued Binary Tree
 */
/**
 * Definition for a binary tree node. public class TreeNode { int val; TreeNode
 * left; TreeNode right; TreeNode(int x) { val = x; } }
 */
class Solution {
    public boolean isUnivalTree(TreeNode root) {
        if (root == null)
            return true;

        return isUnivalTree(root.left, root.val) && isUnivalTree(root.right, root.val);
    }

    private boolean isUnivalTree(TreeNode root, int key) {
        if (root == null)
            return true;

        if (root.val != key)
            return false;
        return isUnivalTree(root.left, key) && isUnivalTree(root.right, key);
    }
}

File: Easy/965.univalued-binary-tree.js
/*
 * @lc app=leetcode id=965 lang=javascript
 *
 * [965] Univalued Binary Tree
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isUnivalTree = function (root) {
  if (root == null) return true;

  if (root.left != null) if (root.val != root.left.val) return false;

  if (root.right != null) if (root.val != root.right.val) return false;

  return isUnivalTree(root.left) && isUnivalTree(root.right);
};
// @lc code=end
